File: src/Parrot.Renderers/IfRenderer.cs
namespace Parrot.Renderers
{
    using System;
    using System.Collections.Generic;
    using Parrot.Renderers.Infrastructure;

    public class IfRenderer : HtmlRenderer
    {
        public IfRenderer(IHost host) : base(host) { }

        public override IEnumerable<string> Elements
        {
            get { yield return "if"; }
        }

        public override void Render(Parrot.Infrastructure.IParrotWriter writer, IRendererFactory rendererFactory, Nodes.Statement statement, IDictionary<string, object> documentHost, object model)
        {
            var localModel = GetLocalModel(documentHost, statement, model);

            if (localModel is bool && (bool)localModel)
            {
                RenderChildren(writer, statement.Children, rendererFactory, documentHost, base.DefaultChildTag, model);
            }
        }
    }
}

File: src/Parrot.Javascript/scripts/Renderers/ifRenderer.ts
///<reference path="../Parser/parser.ts" />
///<reference path="./irenderer.ts" />
///<reference path="./rendererProvider.ts" />
///<reference path="../Infrastructure/ObjectModelValueProvider.ts" />
///<reference path="../Infrastructure/ValueTypeProvider.ts" />
///<reference path="../exceptions.ts" />
///<reference path="./tagBuilder.ts" />
///<reference path="./baseRenderer.ts" />
///<reference path="./selfClosingRenderer.ts" />

class IfRenderer extends HtmlRenderer {
    defaultChildTag: string = "";
    rendererProvider: RendererProvider;
    elements: string[] = ["if"];

    render(statement: Statement, host: any[], model: any, rendererProvider: RendererProvider): string {
        this.rendererProvider = rendererProvider;
        //get the local model
        //var localModel = this.getLocalModelValue(host, statement, model);
        var localModel = this.getLocalModelValue(host, statement, model);

        if (localModel != null && localModel == true) {
            return this.renderChildren(statement, host, model, this.defaultChildTag);
        }
    }
}